packages/contracts/src/contracts/current/protocol/AssetTransferProxy/AssetProxyEncoderDecoder.sol

```diff
@@ -0,0 +1,59 @@
+/*
+
+  Copyright 2018 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.4.21;
+
+import "./LibBytes.sol";
+import "./IAssetProxyEncoderDecoder.sol";
+
+contract AssetProxyEncoderDecoder is
+    LibBytes,
+    IAssetProxyEncoderDecoder
+{
+
+    /// @dev Encodes an order's metadata into a byte array for the respective asset proxy.
+    /// @param assetProxyId Id of the asset proxy.
+    /// @param tokenAddress Address of the asset.
+    /// @return assetMetadata Byte array encoded for the respective asset proxy.
+    function encodeMetadata(
```

Comment:
@abandeali1 We'll be able to remove this function after moving encoded metadata into the order. The settlement layer will simply pass along the encoded metadata right through to the destination proxy.